arbnode/inbox_reader.go

```diff
@@ -279,7 +283,7 @@ func (ir *InboxReader) run(ctx context.Context) error {
 				// nolint:nilerr
 				return nil
 			}
-			if from.Cmp(currentHeight) >= 0 {
+			if from.Cmp(currentHeight) > 0 {
```

Comment:
This is the core bug fixed, that prevented it from scanning the most recent block alone for new inbox messages (but the neededBlockHeight check still passed so it'd try again and do unnecessary queries)